Partials (#8755)
withastro/astro: .changeset/brave-pots-drop.md

---
'astro': minor
---

> Partials

Reviewer: ```suggestion
Page Partials
```

or Partial Pages? or just Partials! Decide how you'd like to colloquially refer to these. I think anything is fine, just want to make sure you've made a conscious choice.

> Any page component can now be identified as a *partial*, allowing you fetch them in the client in order to replace only parts of the page. Partials are used in conjuction with a rendering library, like htmx or Stimulus or even just jQuery.

Reviewer 2: Since I don't have any comments on the code, I'll just want to mention that as a non-native speaker, I stumbled over not reading a "to" before "fetch". 😅

> Pages marked as partials do not have a `doctype` or any head content included in the rendered result. You can mark any page as a partial by setting this option:

Reviewer: ```suggestion
You can mark any `.astro` page as a partial with an export statement:
```

Author: Do you not think it's worth mentioning what this feature does here? I'm thinking in the context of the blog post, shouldn't we explain that it removes the head content?

Reviewer: I'm looking now and one of my comments (my main one) is entirely missing! I rewrote a bunch of stuff to add this earlier!

```astro
---
export const partial = true;
---

<li>This is a single list item.</li>

> ```

Reviewer: ``````suggestion
```

Other valid page files that can export a value (e.g. `.mdx`) can also be marked as partials.

Read more about [Astro page partials](/en/core-concepts/astro-pages/#partials) in our documentation.
`````` 
Not perfect, but something like this?

---
'astro': minor
---

Page Partials

> Any page component can now be identified as a *partial*, allowing you fetch them in the client in order to replace only parts of the page. Partials are used in conjuction with a rendering library, like htmx or Stimulus or even just jQuery.

Reviewer: ```suggestion
A page component can now be identified as a *partial* page, which will render its HTML content without including a `<! DOCTYPE html>` declaration nor any `<head>` content.

A rendering library, like htmx or Stimulus or even just jQuery can access partial content on the client to dynamically update only parts of a page.
```
OK, the original is lost to history... but I was thinking of something *like* this. Again, edit as you like.
